### ISSUE TITLE
Satoshi Endpoint Proxy

### ISSUE THREAD

**Issue by arik-so** [ordinals/ord#4021]
The `server --proxy` option is helpful for testing mainnet inscriptions' behavior locally, but a lot of inscriptions that require such testing rely on a specific dynamically inscribed ordinal, which is usually uninscribed locally, and breaks testability.

I wonder whether allowing the `--proxy` option to also apply as a fallback for the `/r/sat/:sat/at/:index` endpoint might be worth pursuing? That should help with resolving dynamic content.

**Comment by raphjaph** [comment]
Yes, this is a good idea. I even have an issue for this https://github.com/ordinals/ord/issues/3806
